crates/goose/src/providers/provider_registry.rs

```diff
@@ -96,14 +96,25 @@ impl ProviderRegistry {
             })
             .collect();
 
+        let mut config_keys = base_metadata.config_keys.clone();
+
+        // Force the API key to be platform-specific over `OPENAI_API_KEY`
+        if let Some(api_key_index) = config_keys
```

Comment:
can  you remove the comment? code speaks for itself

Reply:
oh in that case, your point doubly stands! if it is wrong then that is really bad (and I guess... as a rule, why comments get stale and can mislead) fair enough! 

Yes, remove the comment! I stand corrected. I assumed it meant what it said which is not great!